observer/baseNodeProvider.go

```diff
@@ -32,6 +36,82 @@ func (bop *baseNodeProvider) initNodesMaps(nodes []*data.NodeData) error {
 	return nil
 }
 
+// ReloadNodes will reload the observers or the full history observers
+func (bop *baseNodeProvider) ReloadNodes(nodesType data.NodeType) data.NodesReloadResponse {
+	bop.mutNodes.RLock()
+	oldNodes := bop.nodes
```

Comment:
you can remove the `oldNodes` variable because you use it just to get the number of nodes.

Reply:
done